### PR TITLE
Updated mlx-regex daemon to use 2M hugepages

### PR DESCRIPTION
**Pull request by gerrygribbon-tic**
mlx-regex daemon will now use 2M hugepages across various Linux distributions instead of using default hugepage size.

### CODE REVIEW

**Review by JamesHunterNV — APPROVED** [review]
Approved, under the assumption that all target OSes have been tested

File: src/mlx5_regex.c [hunk 1]
```diff
@@ -468,6 +468,15 @@ mlx5_regex_register_read(struct ibv_context *ctx, int engine_id,
 	return 0;
 }
 */
+
+/* Check if Linux distribution supports the MAP_HUGE_SHIFT define. */
+/* We will allocate from the 2M hugepage pool.                     */
+#ifdef MAP_HUGE_SHIFT
+#define MLX_REGEX_MAP_HUGE_2MB (21 << MAP_HUGE_SHIFT)
```
Comment (rboyd-nvidia) [inline src/mlx5_regex.c:475]: Should we not use the provided define for this? `#define MAP_HUGE_2MB    (21 << MAP_HUGE_SHIFT)` From: https://man7.org/linux/man-pages/man2/mmap.2.html

I think it's best to not duplicate effort thats already done for us. and then i would also reccommend simplifying the expression by masking the flag early like so:
```
    #ifdef MAP_HUGE_2MB
    #define MLX_REGEX_MAP_HUGETLB (MAP_HUGETLB | MAP_HUGE_2MB)
    #else 
    #define MLX_REGEX_MAP_HUGETLB (MAP_HUGETLB)
    #endif

    ...
    ctx->db_ctx[engine_id].mem_desc.ptr = mmap(NULL, db_size, PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MLX_REGEX_MAP_HUGETLB, -1, 0);
    
```

**Review by rboyd-nvidia — COMMENTED** [review]
One minor comment, up to you if you agree its worth doing. Otherwise looks good.

File: src/mlx5_regex.c (hunk 1 above)
Comment (gerrygribbon-tic) [reply src/mlx5_regex.c:475]: Hi Robert, I'm not 100% sure if the MAP_HUGE_2MB define is available in all OSs - I will merge the commit and come back to this, as no time to retry on all OSs. I have a  vague memory that it wasn't available when investigating the huge pages issue initially, and the MAP_HUGE_SHIFT define was.

**Pull request merged**